src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/jira/Jira.kt

```diff
@@ -1,41 +1,113 @@
 package com.atlassian.performance.tools.awsinfrastructure.api.jira
 
+import com.atlassian.performance.tools.awsinfrastructure.SuppressingRunnable
 import com.atlassian.performance.tools.awsinfrastructure.api.RemoteLocation
 import com.atlassian.performance.tools.concurrency.api.submitWithLogContext
 import com.atlassian.performance.tools.infrastructure.api.MeasurementSource
 import com.atlassian.performance.tools.infrastructure.api.jvm.jmx.JmxClient
-import com.atlassian.performance.tools.jvmtasks.api.TaskTimer.time
 import com.google.common.util.concurrent.ThreadFactoryBuilder
 import org.apache.logging.log4j.LogManager
 import org.apache.logging.log4j.Logger
 import java.net.URI
 import java.util.concurrent.Executors
 
-class Jira(
+/**
+ * @param extraResultsSources source of results/diagnostics of: reverse proxy, Crowd, LDAP, DVCS, DB, Jira plugins
+ * or anything that can be integrated with Jira as part of web application provisioning
+ */
+class Jira private constructor(
     private val nodes: List<StartedNode>,
     val jiraHome: RemoteLocation,
     val database: RemoteLocation?,
     val address: URI,
-    val jmxClients: List<JmxClient> = emptyList()
+    val jmxClients: List<JmxClient>,
+    private val extraResultsSources: List<MeasurementSource>
```

Comment:
Sggestion from pairing with @ewefie: `extraMeasurementSource`

and propagate to all new API